test/xebow_test.exs

```diff
@@ -3,31 +3,223 @@ defmodule XebowTest do
 
   alias RGBMatrix.Animation
 
-  defmodule MockAnimations.Type1 do
-    use Animation
+  setup_all [
+    :create_mock_animations,
+    :create_single_animation,
+    :create_single_schema
+  ]
 
-    @impl Animation
-    def new(_leds, _config) do
-      nil
+  test "has layout" do
+    assert %Layout{} = Xebow.layout()
+  end
+
+  describe "can get and set active animation types" do
+    test "with a list of animation modules", %{
+      mock_animations: mock_animations
+    } do
+      assert Xebow.set_active_animation_types(mock_animations) == :ok
+      assert Xebow.get_active_animation_types() == mock_animations
     end
 
-    @impl Animation
-    def render(_state, _config) do
-      {1000, [], nil}
+    test "with an empty list" do
+      assert Xebow.set_active_animation_types([]) == :ok
+      assert Xebow.get_active_animation_types() == []
     end
   end
 
-  test "has layout" do
-    assert %Layout{} = Xebow.layout()
+  describe "animations can be cycled" do
+    setup %{mock_animations: mock_animations} do
+      Xebow.set_active_animation_types(mock_animations)
+    end
+
+    test "forward" do
+      assert Xebow.next_animation() == :ok
+    end
+
+    test "backward" do
+      assert Xebow.previous_animation() == :ok
+    end
+
+    test "through the entire list, repeatedly", %{
```

Comment:
ℹ️ (optional)

Another way to describe these requirements would be that the animation list wraps. So rather than scrolling through the whole list a couple times, the critical part to test becomes the first element connecting to the last, and the last to the first, depending on the cycle direction.

Reply:
There's no way to guarantee where in the list we are at any given time, though, so it's got to go through length+1 at minimum, and *2 was where I ended at.

If I can think of something that reads better, I'll update the name of the test.

Reply:
>There's no way to guarantee where in the list we are

`List.first` and `List.last`, and call `Xebow.next_animation` or `Xebow.previous_animation` respectively?

Reply:
Yeah, but doesn't that kind of lean toward testing internal implementation? This part of the application shouldn't care whether it's stored as a list, or a map, or some opaque data structure that runs through NIFs.

We're passing in a list of animations to use as active, and we know the application make it possible to cycle through all of them. We shouldn't care if they'll get loaded in the same order they're provided, or if it's sorted at some point, or randomized, or something completely undefined.

I actually had a previous version of this test that cycled through the given list using `next`/`previous`, then found the name of the animation module by getting the config, and then deleted that from the list as part of the test, then asserted the result was an empty list, after doing this `count` times. It seemed like... really not a very good way of doing it, at the time, which is why I got rid of it. I could go back to that method.

Reply:
I believe previous/next are going away in an upcoming UI change, so feel free to proceed with these tests in whatever way you think makes the most sense.